### PR TITLE
Dev stage 4

### LINKED ISSUE #6: Xavier weights intialization

### LINKED ISSUE #13: Numpy true_divide at Softmax function produces nan values

### PR DESCRIPTION
**Pull request by kuweg**

### CODE REVIEW

**Comment by kuweg** [comment]
PureNN able to perform MNIST classification with ~97% accuracy on test set. 
Added `softmax` and `CrossEntropyLoss` support
Fancy `.info()` for model's description

TODO:
1. Perform `regression` tests
2. Add dataloader for `boston_housing`
